test/e2e/framework/deployment.go

```diff
@@ -28,6 +28,26 @@ func CreateDeployment(client kubernetes.Interface, deployment *appsv1.Deployment
 	})
 }
 
+// UpdateDeployment updates the Deployment object.
+func UpdateDeployment(client kubernetes.Interface, deployment *appsv1.Deployment) {
+	ginkgo.By(fmt.Sprintf("Update Deployment(%s/%s)", deployment.Namespace, deployment.Name), func() {
+		gomega.Eventually(func() error {
+			_, err := client.AppsV1().Deployments(deployment.Namespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+			return err
```

Comment:
I am afraid it will always fail if this resource is updated by others

Reply:
@jameszhangyukun How about adding the query operation before the update?